data/virt_autotest/setup_dns_service.txt

```diff
@@ -0,0 +1,252 @@
+#zone "123.168.192.in-addr.arpa" in {
+#        type master;
+#        file "123.168.192.zone";
+#};
+awk -v dnsvar=${dns_domain_reverse} 'done != 1 && /^zone.*$/ { print "zone \""dnsvar".in-addr.arpa\" in {\n        type master;\n        file \""dnsvar".zone\";\n};"; done=1 } 1' ${dns_config_file} > ${dns_config_file_tmp}
+mv ${dns_config_file_tmp} ${dns_config_file}
+echo -e "\n${dns_domain_reverse} zone file content:\n" | tee -a ${setup_log_file}
+cat ${dns_reverse_zone_file} |  tee -a ${setup_log_file}
+
+dns_resolv_file="/etc/resolv.conf"
+dns_resolv_file_tmp="/etc/resolv.conf.tmp"
+cp ${dns_resolv_file} ${dns_resolv_file}.orig
+awk -v dnsvar=${bridgeip} 'done != 1 && /^nameserver.*$/ { print "nameserver "dnsvar"\n"; done=1 } 1' ${dns_resolv_file} > ${dns_config_file_tmp}
+mv ${dns_config_file_tmp} ${dns_resolv_file}
+sed -irn "/^search/ s/$/ ${dns_domain_forward}/" ${dns_resolv_file}
+get_nameservers=`cat ${dns_resolv_file} | grep -iE "^nameserver" | awk '{print $NF}'`
+nameservers_array=$(echo -e ${get_nameservers})
+unset forwarders_array
+declare -a forwarders_array=""
+for single_nameserver in ${nameservers_array[@]};do
+	forwarders_array+=(${single_nameserver}\;)
+done	
+sed -irn "s/forwarders.*$/forwarders { `echo -e ${forwarders_array[@]}` };/" ${dns_config_file}
+
+dns_service_name="named"
+get_os_installed_release=`lsb_release -r | grep -oE "[[:digit:]]{2}"`
+os_installed_release=$(echo ${get_os_installed_release})
+if [[ ${os_installed_release} -gt '11' ]];then
+	systemctl enable ${dns_service_name}
+	systemctl restart ${dns_service_name}
+	dns_service_failed=$(echo $?)
+	systemctl status ${dns_service_name} | tee -a ${setup_log_file}
+else
+	service ${dns_service_name} restart
+	dns_service_failed=$(echo $?)
+	service ${dns_service_name} status | tee -a ${setup_log_file}
+fi
+if [[ ${dns_service_failed} -ne 0 ]];then
+	echo -e "DNS service did not start up as normal. Please investigate.\n" | tee -a ${setup_log_file}
+	echo -e "DNS resolver file content:\n" | tee -a ${setup_log_file}
+	cat ${dns_resolv_file} | tee -a ${setup_log_file}
+	echo -e "NAMED service file content:\n" | tee -a ${setup_log_file}
+	cat ${dns_config_file} | tee -a ${setup_log_file}
+	exit 1
+fi
+
+sleep 45s
+failed_count=0
+for vmhost in ${vm_hostnames_array[@]};do
+        ping -c 5 ${vmhost}
+	if [[ `echo $?` -ne 0 ]];then
+		echo -e "Connection to ${vmhost} using its dns name failed !\n" | tee -a ${setup_log_file}
+	    	failed_count=$((${failed_count} + 1))
+        else
+            	echo -e "Connection to ${vmhost} using its dns name succeeded !\n" | tee -a ${setup_log_file}
+        fi
+done
+
+chmod 777 ${setup_log_file}
+if [[ ${failed_count} -gt 0 ]];then
+	echo -e "DNS service does not work for at least one vm. Please pay attention !\n" | tee -a ${setup_log_file}
+	exit 1
+else
+	echo -e "SSH Connection to all virtual machines by using DNS names is working now.\n" | tee -a ${setup_log_file}
+fi
```

Comment:
Please add some comments to above ~200 lines to explain what the sections of code are doing logically. This will help others know the code logic.

Reply:
Done